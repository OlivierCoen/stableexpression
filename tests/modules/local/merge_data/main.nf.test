nextflow_process {

    name "Test Process MERGE_DATA"
    script "modules/local/merge_data/main.nf"
    process "MERGE_DATA"
    tag "merge"
    tag "module"

    test("Merge counts - filter out genes with zero counts") {

        when {
            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/merge_data/input/counts*.parquet', checkIfExists: true).collect()
                ch_designs = Channel.fromPath( '$projectDir/tests/test_data/merge_data/input/design*.csv', checkIfExists: true).collect()
                ch_dataset_stat = Channel.fromPath( '$projectDir/tests/test_data/merge_data/input/dataset_stat*.csv', checkIfExists: true).collect()
                nb_candidate_genes = 3
                input[0] = ch_counts
                input[1] = ch_designs
                input[2] = ch_dataset_stat
                input[3] = nb_candidate_genes
                """
            }

        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.all_counts[0]).exists() },
                { with( path(process.out.all_designs[0]).csv( header: true ) ) {
                    assert columnNames == ["batch", "condition", "sample"]
                    assert rowCount == 27
                }},
                { assert path(process.out.candidate_gene_counts[0]).exists() },
                { with( path(process.out.distribution_correlations[0]).csv( header: false ) ) {
                    assert rowCount == 27
                    assert columnCount == 2
                }},
                { with( path(process.out.gene_count_statistics[0]).csv( header: false ) ) {
                    assert rowCount == 27
                    assert columnCount == 2
                }},
                { with( path(process.out.skewness_statistics[0]).csv( header: false ) ) {
                    assert rowCount == 27
                    assert columnCount == 2
                }},
                { with( path(process.out.ks_test_statistics[0]).csv( header: false ) ) {
                    assert rowCount == 27
                    assert columnCount == 2
                }}
            )


        }

    }

}
