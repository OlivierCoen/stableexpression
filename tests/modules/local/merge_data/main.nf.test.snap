{
    "Merge counts - filter out genes with zero counts": {
        "content": [
            {
                "0": [
                    "all_counts.parquet:md5,cc26c33868c694c06e45338c1b4d0a77"
                ],
                "1": [
                    "all_designs.csv:md5,32614fc915e7f2748f4c2f749bb688ec"
                ],
                "2": [
                    "gene_count_statistics.csv:md5,5aef723f917dfeb7212cf4ccd82f8866"
                ],
                "3": [
                    "skewness_statistics.csv:md5,5aef723f917dfeb7212cf4ccd82f8866"
                ],
                "4": [
                    "ks_test_statistics.csv:md5,fca4e42eab03c715ed58fa736a81bb50"
                ],
                "5": [
                    "candidate_gene_counts.parquet:md5,bd82490ae5db9d510090a0b271e74710"
                ],
                "6": [
                    "distribution_correlations.csv:md5,7bb8292d69a32077356b5b16afb76245"
                ],
                "7": [
                    [
                        "MERGE_DATA",
                        "python",
                        "3.12.8"
                    ]
                ],
                "8": [
                    [
                        "MERGE_DATA",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "all_counts": [
                    "all_counts.parquet:md5,cc26c33868c694c06e45338c1b4d0a77"
                ],
                "all_designs": [
                    "all_designs.csv:md5,32614fc915e7f2748f4c2f749bb688ec"
                ],
                "candidate_gene_counts": [
                    "candidate_gene_counts.parquet:md5,bd82490ae5db9d510090a0b271e74710"
                ],
                "distribution_correlations": [
                    "distribution_correlations.csv:md5,7bb8292d69a32077356b5b16afb76245"
                ],
                "gene_count_statistics": [
                    "gene_count_statistics.csv:md5,5aef723f917dfeb7212cf4ccd82f8866"
                ],
                "ks_test_statistics": [
                    "ks_test_statistics.csv:md5,fca4e42eab03c715ed58fa736a81bb50"
                ],
                "skewness_statistics": [
                    "skewness_statistics.csv:md5,5aef723f917dfeb7212cf4ccd82f8866"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-08T14:36:15.901788378"
    }
}